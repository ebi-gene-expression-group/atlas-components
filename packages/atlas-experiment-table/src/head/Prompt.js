import React from 'react'
import PropTypes from 'prop-types'

class Prompt extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedFileTypes: props.downloadFileTypes.map(fileType => fileType.id)
    }

    this.onClick = this.onClick.bind(this)
  }

  onClick(selectedFileType) {
    const selectedFileTypes = [...this.state.selectedFileTypes]
    this.setState({
      selectedFileTypes: selectedFileTypes.includes(selectedFileType) ?
        selectedFileTypes.filter(fileType => fileType !== selectedFileType) :
        [selectedFileType, ...selectedFileTypes]
      },
      () => this.props.onSelect(this.state.selectedFileTypes)
    )
  }

  render() {
    return <div>
        <p style={{paddingBottom: `1rem`}}>Choose file types...</p>
          {
            this.props.downloadFileTypes.map((fileType, idx) =>
              <div key={`filetype${idx}`} style={{display:"flex"}}>
                <input type={`checkbox`} className={`checkbox`} id={fileType.id}
                       checked={this.state.selectedFileTypes.includes(fileType.id)}
                       onChange={() => this.onClick(fileType.id)}/>
                <p id={fileType.description} style={{paddingLeft: `1rem`, paddingBottom: `1rem`}}>
                  {fileType.description}
                </p>
              </div>
            )
          }

      </div>

  }
}

Prompt.propTypes = {
  onSelect: PropTypes.func.isRequired,
  downloadFileTypes: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      id: PropTypes.string
    })
  ).isRequired
}

export default Prompt
