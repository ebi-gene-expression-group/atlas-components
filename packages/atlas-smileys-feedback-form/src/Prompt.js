import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import smileyData from './smileyData'
import SmileyDiv from './SmileyDiv'

const SmileyContainer = styled.div`
  display: flex;
  justify-content: space-evenly;
`

class Prompt extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedSmileyScore: 0
    }

    this.onClick = this.onClick.bind(this)
  }

  onClick(selectedSmileyScore) {
    this.setState({
      selectedSmileyScore: selectedSmileyScore
    })
    this.props.onSelect(selectedSmileyScore)
  }

  render() {
    return(
      <div>
        <p style={{paddingBottom: `1rem`}}>How satisfied are you?</p>
        <SmileyContainer>
          {
            smileyData.map((smiley, idx) =>
              <SmileyDiv key={idx} onClick={() => this.onClick(smiley.score)}
                emoji={smiley.emoji}
                label={smiley.label}
                selected={smiley.score === this.state.selectedSmileyScore}/>
            )
          }
        </SmileyContainer>
        <p style={{paddingTop: `1rem`}}>
          <a href={this.props.feedbackFormLink} target={`_blank`}>Click here if you need support.</a>
        </p>
      </div>
    )
  }
}

Prompt.propTypes = {
  onSelect: PropTypes.func.isRequired,
  feedbackFormLink: PropTypes.string.isRequired
}

export default Prompt
