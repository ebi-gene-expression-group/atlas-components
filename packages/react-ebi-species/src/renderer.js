"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
