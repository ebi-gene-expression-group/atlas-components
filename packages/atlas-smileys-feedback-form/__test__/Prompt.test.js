import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Prompt from '../src/Prompt'
import { Emoji } from 'emoji-mart'

Enzyme.configure({ adapter: new Adapter() })


describe(`Prompt`, () => {

  test(`should render 5 smiley faces and 1 input text box`, () => {
    expect(shallow(<Prompt />).find(Emoji)).toHaveLength(5)
    expect(shallow(<Prompt />).find(`input`)).toHaveLength(1)
  })

  test(`should display a feedback scale when click smiley faces`, () => {
    const wrapper = mount(<Prompt onSelect={()=>{}} />)
    expect(wrapper.find(`#scale`).text()).toBe(`empty`)

    wrapper.find(Emoji).first().simulate(`click`)
    expect(wrapper.find(`#scale`).text()).toBe(`Terrible`)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<Prompt />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
