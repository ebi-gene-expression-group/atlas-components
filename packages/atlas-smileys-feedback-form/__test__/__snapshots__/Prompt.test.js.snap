// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt matches snapshot 1`] = `
<div>
  <p
    style={
      Object {
        "paddingBottom": "1rem",
      }
    }
  >
    How satisfied are you?
  </p>
  <div
    className="sc-bwzfXH bXowzj"
  >
    <div
      className="sc-bdVaJa HHxyE"
      label="Terrible"
      onClick={[Function]}
      selected={false}
    />
    <div
      className="sc-bdVaJa crhCQw"
      label="Bad"
      onClick={[Function]}
      selected={false}
    />
    <div
      className="sc-bdVaJa iyzqPF"
      label="Okay"
      onClick={[Function]}
      selected={false}
    />
    <div
      className="sc-bdVaJa dMybav"
      label="Good"
      onClick={[Function]}
      selected={false}
    />
    <div
      className="sc-bdVaJa fRkOYZ"
      label="Great"
      onClick={[Function]}
      selected={false}
    />
  </div>
  <p
    style={
      Object {
        "paddingTop": "1rem",
      }
    }
  >
    <a
      href="https://www.ebi.ac.uk/support/gxasc"
      target="_blank"
    >
      Click here if you need support.
    </a>
  </p>
</div>
`;
